racket/collects/openssl/mzssl.rkt

```diff
@@ -646,15 +646,35 @@ TO DO:
                     1))))
           ssl-listener pathname))
 
-(define (ssl-load-private-key! ssl-context-or-listener pathname
+(define (ssl-load-private-key! ssl-context-or-listener pathname-or-bytes
                                [rsa? #t] [asn1? #f])
+  (if (bytes? pathname-or-bytes)
+    (let ([bio (BIO_new_mem_buf pathname-or-bytes (bytes-length pathname-or-bytes))]
```

Comment:
The `bio` should be opened after the `ctx` is retrieved; otherwise if the context is bad, an error is raised outside of the `with-failure` block and the `bio` doesn't get freed.

The `bio` also needs to be freed on success. It looks like `ssl-server-context-enable-dhe!` also creates a memory bio that doesn't get freed, and maybe you were using that as a guide, but as far as I can tell that function is wrong too.